sdk/tests/conformance2/renderbuffers/readbuffer.html

```diff
@@ -0,0 +1,192 @@
+  gl.bindTexture(gl.TEXTURE_2D, tex);
+  gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, canvas.width, canvas.height, 0,
+                      gl.RGBA, gl.UNSIGNED_BYTE, null);
+  gl.copyTexImage2D(gl.TEXTURE_2D, 0, gl.RGBA, 0, 0, canvas.width, canvas.height, 0);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+      "should be no errors when reading from GL_BACK on the default framebuffer.");
+  testPixels(0, 0, backColor);
+
+  // switch to fbo from default fb, read buffer will switch to GL_COLOR_ATTACHMENT0 from GL_BACK by default
+  var fb = gl.createFramebuffer();
+  gl.bindFramebuffer(gl.FRAMEBUFFER, fb);
+  var colorbuffer = gl.createRenderbuffer();
+  gl.bindRenderbuffer(gl.RENDERBUFFER, colorbuffer);
+  gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.RENDERBUFFER, colorbuffer);
+  gl.renderbufferStorage(gl.RENDERBUFFER, gl.RGBA8, canvas.width, canvas.height);
+
+  shouldBe('gl.getParameter(gl.READ_BUFFER)', 'gl.COLOR_ATTACHMENT0');
+  var fboColor = [0, 255, 0, 255];
+  gl.clearColor(fboColor[0] / 255, fboColor[1] / 255, fboColor[2] / 255, fboColor[3] / 255);
+  gl.clear(gl.COLOR_BUFFER_BIT);
+  gl.copyTexImage2D(gl.TEXTURE_2D, 0, gl.RGBA, 0, 0, canvas.width, canvas.height, 0);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+      "should be no errors when reading from GL_COLOR_ATTACHMENT0 on fbo.");
+  testPixels(0, 0, fboColor);
+
+  // test on the fbo, set read buffer to GL_COLOR_ATTACHMENT1
+  gl.readBuffer(gl.BACK);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION,
+      "calling readBuffer with GL_BACK on fbo should generate INVALID_OPERATION.");
+
+  gl.readBuffer(gl.NONE);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+      "calling readBuffer with GL_NONE on fbo should succeed.");
+  gl.readBuffer(gl.COLOR_ATTACHMENT0);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+      "calling readBuffer with GL_COLOR_ATTACHMENT0 on fbo should succeed.");
+
+  var maxColorAttachments = gl.getParameter(gl.MAX_COLOR_ATTACHMENTS);
+  gl.readBuffer(gl.COLOR_ATTACHMENT0 + maxColorAttachments);
+  wtu.glErrorShouldBe(gl, gl.INVALID_ENUM,
+      "calling readBuffer with GL_COLOR_ATTACHMENTi that exceeds MAX_COLOR_ATTACHMENT on fbo should generate INVALID_ENUM.");
+  gl.readBuffer(gl.COLOR_ATTACHMENT1);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+      "calling readBuffer with GL_COLOR_ATTACHMENT1 on the fbo should succeed.");
+  shouldBe('gl.getParameter(gl.READ_BUFFER)', 'gl.COLOR_ATTACHMENT1');
+
+  gl.copyTexImage2D(gl.TEXTURE_2D, 0, gl.RGBA, 0, 0, canvas.width, canvas.height, 0);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION,
+      "should generate INVALID_OPERATION when reading from GL_COLOR_ATTACHMENT1 but this attachment has no image currently.");
+
+  var colorbuffer1 = gl.createRenderbuffer();
+  gl.bindRenderbuffer(gl.RENDERBUFFER, colorbuffer1);
+  gl.renderbufferStorage(gl.RENDERBUFFER, gl.RGBA8, canvas.width, canvas.height);
+  gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT1, gl.RENDERBUFFER, colorbuffer1);
```

Comment:
the status of framebuffer is incomplete, so several tests below will fail in Chromium with my patch. Is there anything wrong in this test?